### PR TITLE
General maintenance

### PR DESCRIPTION
**Pull request by RalphSteinhagen**
maintenance in response to issues found by static code review tools (Coverity, PMD, ...)

### CODE REVIEW

File: chartfx-math/src/main/java/de/gsi/math/spectra/Convolution.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ private void init(final int size) {
                 if (i < offset) {
                     // match ZOH at lower boundary
                     fdata[i] = data[0];
-                } else if (i >= offset && i < offset + data.length) {
+                } else if (i < offset + data.length) {
                     fdata[i] = data[i - offset];
                 } else if (i > offset + data.length) {
                     // match ZOH at upper boundary
```
Comment (wirew0rm) [inline chartfx-math/src/main/java/de/gsi/math/spectra/Convolution.java]: What happens for `i == offset + data.length` ?

File: chartfx-dataset/src/main/java/de/gsi/dataset/spi/DataSetBuilder.java [hunk 2]
```diff
@@ -233,6 +237,7 @@ protected DataSet buildRawDataSet(final String dsName) {
             final int size = initialCapacity < 0 ? minArrays : Math.min(minArrays, initialCapacity);
             dataSet = buildWithYErrors(dsName, size);
         }
+
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-dataset/src/main/java/de/gsi/dataset/spi/DataSetBuilder.java]: done see commit: https://github.com/GSI-CS-CO/chart-fx/pull/18/commits/91982f4ea5d6bbd39542be4cf89f43573c10802f

File: chartfx-math/src/main/java/de/gsi/math/filter/iir/BandStopTransform.java [hunk 3]
```diff
@@ -88,22 +88,22 @@ private ComplexPair transform(Complex c) {
...
         u = u.add(a);
-        u = MathSupplement.addmul(u, -a, c);
+        u = u.add(c.multiply(-a));
 
         Complex d = new Complex(b + 1);
-        d = MathSupplement.addmul(d, b - 1, c);
+        d = d.add(c.multiply(b - 1));
 
         return new ComplexPair(u.divide(d), v.divide(d));
     }
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-math/src/main/java/de/gsi/math/filter/iir/BandStopTransform.java]: specific lines contained an error that has been fixed with commit: https://github.com/GSI-CS-CO/chart-fx/pull/18/commits/91982f4ea5d6bbd39542be4cf89f43573c10802f

File: chartfx-math/src/main/java/de/gsi/math/filter/iir/BandPassTransform.java [hunk 4]
```diff
@@ -24,15 +24,14 @@
 import org.apache.commons.math3.complex.Complex;
 
 /**
- * Transforms from an analogue bandpass filter to a digital bandstop filter
+ * Transforms from an analogue bandpass filter to a digital band-stop filter
  */
 public class BandPassTransform {
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-math/src/main/java/de/gsi/math/filter/iir/BandPassTransform.java]: true. this was already present in the existing/original code...

as a side remark: these classes also overwrite the calling method parameter. In most cases, it seems that the modified method parameter isn't further used (ie. could be marked final) but I found some cases where the changing of the passed parameter is needed. This deserves a refactoring ... but probably at a later stage ... unless you find the time

File: chartfx-math/src/main/java/de/gsi/math/filter/iir/ChebyshevII.java [hunk 5]
```diff
@@ -121,46 +113,37 @@ public void lowPass(final int order, final double sampleRate, final double cutof
...
-        new HighPassTransform(cutoffFrequency / sampleRate, m_digitalProto, m_analogProto);
+        new HighPassTransform(cutoffFrequency / sampleRate, digitalProto, analogProto);
 
-        setLayout(m_digitalProto, directFormType);
+        setLayout(digitalProto, directFormType);
     }
 
     /**
-     * ChebyshevI Highpass filter with default topology
+     * ChebyshevI High-pass filter with default topology
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-math/src/main/java/de/gsi/math/filter/iir/ChebyshevII.java]: fixed with https://github.com/GSI-CS-CO/chart-fx/pull/18/commits/8121abd11f74a3b969cd57644c7f738cb3455bba

File: chartfx-math/src/main/java/de/gsi/math/filter/iir/HighPassTransform.java [hunk 6]
```diff
@@ -33,7 +33,7 @@
     public HighPassTransform(final double fc, final LayoutBase digital, final LayoutBase analog) {
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-math/src/main/java/de/gsi/math/filter/iir/HighPassTransform.java]: as above ... would change interface... this commits are primarily syntax/style only

File: chartfx-math/src/main/java/de/gsi/math/filter/iir/LowPassTransform.java [hunk 7]
```diff
@@ -47,7 +47,7 @@ private Complex transform(Complex c) {
     public LowPassTransform(final double fc, final LayoutBase digital, final LayoutBase analog) {
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-math/src/main/java/de/gsi/math/filter/iir/LowPassTransform.java]: as above

File: chartfx-math/src/main/java/de/gsi/math/fitter/GaussFitting.java [hunk 8]
```diff
@@ -13,11 +13,12 @@
 
     /**
      * fit precise bunch width and location based on peak indication and width estimate
```
[the remark this answers is not in the document]
Comment (RalphSteinhagen) [reply chartfx-math/src/main/java/de/gsi/math/fitter/GaussFitting.java]: changed with https://github.com/GSI-CS-CO/chart-fx/pull/18/commits/35910f42219f1fc9cf9ce5811ed0b38b026ad6a4

**Pull request merged**